src/ops/view.rs

```diff
@@ -12,6 +15,150 @@ use cargo::{
 
 use super::style::{ERROR, HEADER, LITERAL, NOP, NOTE, WARN};
 
+#[allow(unused)]
+#[derive(PartialEq, Eq)]
+enum Hyperlink<'a> {
```

Comment:
Through `GlobalContext` you can get a `Shell` which has hyperlink support

Reply:
Okay~ Give me some seconds to investigate~

Reply:
Hi Ed~

Yeah, the `GlobalContext` had owned the hyperlink supports, e.g., `out_hyperlink` and `err_hyperlink`, but it can only let the URL itself be a hyperlink, cannot accept the customized text(I will use the HTML/Javascript snippets as the examples):

`GlobalContext` only supports: `<a href="${url}">${url}</a>`.
The `Hyperlink` struct here can supports: `<a href="${url}">${customized_text}</a>`.

Reply:
Hi Ed~

I had used the `GlobalContext` to determine that the current terminal or `GlobalContext.shell` supports hyperlink or not.

Reply:
Hi Ed, I got what you mean.

https://github.com/rust-lang/cargo/blob/a8d72c675ee52dd57f0d8f2bae6655913c15b2fb/src/cargo/core/compiler/job_queue/mod.rs#L832

I'll change the implementation by following the codes above.

Reply:
Why are you trying to construct it directly, rather than through `out_hyperlink`?  What is the immutable/mutable problem?

Reply:
Hi Ed~

Thanks for your patience sincerely.

I'm just wondering not to change the existed `fn`s parameters, such as `print_deps` and `print_deps`...

My bad. I was stuck into my own thoughts.